editor/editor_resource_picker.cpp

```diff
@@ -216,7 +218,13 @@ void EditorResourcePicker::_update_menu_items() {
 		}
 
 		if (is_editable()) {
-			edit_menu->add_icon_item(get_editor_theme_icon(SNAME("Clear")), TTR("Clear"), OBJ_MENU_CLEAR);
+			if (_owner && _owner->has_meta("__custom_type_script")) {
```

Comment:
You are not checking whether the current resource is script. I think this should be handled in property setup instead.